pkg/scheduler/internal/cache/debugger/dumper.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debugger
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"strings"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/kubernetes/pkg/scheduler/cache"
+	internalcache "k8s.io/kubernetes/pkg/scheduler/internal/cache"
+	"k8s.io/kubernetes/pkg/scheduler/internal/queue"
+)
+
+type CacheDumper struct {
+	cache    internalcache.Cache
+	podQueue queue.SchedulingQueue
+}
+
+func (d *CacheDumper) DumpAll() {
+	d.DumpNodes()
+	d.DumpSchedulingQueue()
+}
+
+// DumpNodes writes NodeInfo to the scheduler logs.
+func (d *CacheDumper) DumpNodes() {
+	snapshot := d.cache.Snapshot()
+	glog.Info("Dump of cached NodeInfo")
+	for _, info := range snapshot.Nodes {
+		glog.Info(PrintNodeInfo(info))
+	}
+}
+
+func PrintNodeInfo(n *cache.NodeInfo) string {
```

Comment:
Maybe `PrintNodeInfo` does not need to be exposed？

Reply:
Yes. Makes sense. Done.